### PR TITLE
Use package.json and CDN to entirely manage front-end dependencies regardless of environment #6346

### LINKED ISSUE #6346: Use package.json and CDN to entirely manage front-end dependencies regardless of environment
It will be a contradiction to #3923, but I have experienced that having to sync the copies (particularly during updates) adds a non-trivial maintenance burden. I therefore propose to reverse the decision made in that issue and opt for using CDN fully for external files, dev environment or not.

Cons:
- Internet connection is required. One of the original intention of that issue was to severe the need for Internet, but hey everyone's on Internet these days.
- If the CDN is down, so is the dev environment, but it will also affect the live environment so the dev environment getting affected is a secondary problem. 😛
- May slow down (or even speed up?) the tests. I can't provide any quantitative evidence of this, though.

Pros:
- Adding/updating/deleting libraries are as simple as adding/updating/deleting entries to/in/from `package.json` (of course, discounting other stuffs such as dependency hell).
- It severs the need to host the local copies of the libraries anymore; reduces repo size and maintenance burden significantly. This burden comes from having to know which files of the library need to be hosted, and affects all three processes of add/update/delete.

The pros far outweigh the cons for me, but that's just one person's (biased) opinion.
@damithc @junhaoyap will need your inputs on this!

### PR DESCRIPTION
**Pull request by wkurniawan07**
Fixes #6346 

The downside of this design is that the test Javascript files have to be put in the production folder (although they will be restricted to admins).

### CODE REVIEW

**Comment by junhaoyap** [comment]
Updating branch, will take a look at a coffee place in the evening

File: src/main/webapp/dev/allJsUnitTests.jsp [hunk 1]
```diff
@@ -201,8 +165,8 @@
             </div>
             <div class="table table-responsive table-striped table-bordered">
                 <a class="btn btn-default btn-xs pull-right pull-down course-enroll-for-test"
-                    href="<%=data.getInstructorCourseEnrollLink(courseDetails.course.id)%>"
-                    title="<%=Const.Tooltips.COURSE_ENROLL%>"
+                    href=""
```
Comment (junhaoyap) [inline src/main/webapp/dev/allJsUnitTests.jsp]: These changes seem unrelated, any comments on them?

File: src/main/webapp/dev/allJsUnitTests.jsp [hunk 2]
```diff
@@ -252,8 +216,8 @@
...
```
Comment (junhaoyap) [inline src/main/webapp/dev/allJsUnitTests.jsp]: Ditto

File: src/main/webapp/dev/allJsUnitTests.jsp (hunk 1 above)
Comment (wkurniawan07) [inline src/main/webapp/dev/allJsUnitTests.jsp]: Yes. It has to do with this file becoming a JSP, thus this previously-innocent looking block becomes something parseable, and unfortunately we do not (nor we need to) have a variable named `data` or importing `Const`.
This is a symptom of copy-pasting from JSP file without checking :p

Comment (junhaoyap) [inline src/main/webapp/dev/allJsUnitTests.jsp]: Ahhah! Okay thanks for disambiguating

**Comment by junhaoyap** [comment]
Ping, ready to go

**Comment by junhaoyap** [comment]
@wkurniawan07 I have updated the branch, what is the merge procedure now, squash and merge via GitHub's UI?

**Comment by wkurniawan07** [comment]
@junhaoyap yup, still the same. I'm actually waiting for #6348 though, because easier (and faster) for me to adapt.

**Comment by junhaoyap** [comment]
@wkurniawan07 okay then so I won't be itchy fingers and merge this while we wait for that ;)

**Comment by wkurniawan07** [comment]
I updated the issue and PR title a bit because it's no longer "Consider" :p
